### PR TITLE
Add support for globstars in standard finder

### PR DESCRIPTION
**Pull request by cbowman0**
Changes from @mark-5 from #1178 

* Manually merge into finders/standard.py.
* Adjust the test cases.

### CODE REVIEW

**Comment by cbowman0** [comment]
I tested on my backup cluster and this appears to work as expected.  I did have some timeouts when I searched across all 9 of the backend servers in the cluster for something that would return a very large result set, but I think that's to be expected.

@obfuscurity do you want to try this out before we merge?

File: webapp/content/js/completer.js [hunk 1]
```diff
@@ -41,7 +41,9 @@ MetricCompleter = Ext.extend(Ext.form.ComboBox, {
   },
 
   prepareQuery: function (queryEvent) {
-    queryEvent.query += '*';
+    if (queryEvent.query.substr(-1) != '*') {
```
Comment (obfuscurity) [inline webapp/content/js/completer.js]: nice

File: webapp/graphite/finders/standard.py [hunk 2]
```diff
@@ -80,8 +82,15 @@ def _find_paths(self, current_dir, patterns):
     else:
       entries = [ pattern ]
 
-    subdirs = [entry for entry in entries if isdir(join(current_dir, entry))]
-    matching_subdirs = match_entries(subdirs, pattern)
+    if using_globstar:
+        matching_subdirs = map(operator.itemgetter(0), os.walk(current_dir))
```
Comment (obfuscurity) [inline webapp/graphite/finders/standard.py]: Shouldn't we check for `isdir` here too?

Comment (cbowman0) [inline webapp/graphite/finders/standard.py]: I tested independently and only saw directories show up.

Output is only directories:
```
$ python2.7 test.py | xargs -L1 -I{}  stat -c%F {} | uniq -c
  27177 directory
```

It did take 1.1 seconds to just run the osdir on this host, but it is at about 65% IO utilization all the time.
```
$ time python2.7 test.py  | wc -l
27177

real	0m1.149s
user	0m0.675s
sys	0m0.478s
```

Test script:
```
$ cat test.py
import os
import operator

current_dir='/opt/graphite/storage/whisper/carbon/'

matching_subdirs = map(operator.itemgetter(0), os.walk(current_dir))

for i in matching_subdirs:
  print i
```

**Comment by obfuscurity** [comment]
👍

File: webapp/graphite/finders/standard.py (hunk 2 above)
Comment (mark-5) [inline webapp/graphite/finders/standard.py]: According to the [docs](https://docs.python.org/3/library/os.html#os.walk), os.walk returns a list of tuples like `(dirpath, dirnames, filenames)`, so I figured that first item we are getting is always a dir, and doesn't need an `isdir` check.

Comment (obfuscurity) [inline webapp/graphite/finders/standard.py]: Ah ok, missed that.

**Pull request merged**